docs/rfcs/0007-deployment-flow.md

```diff
@@ -0,0 +1,111 @@
+- Start Date: 2021-11-03
+- Target Version: 0.22.0
+
+# Summary
+This RFC introduces a new way to enable users to deploy their complex system (or applications) in a more flexible way by manipulating the currently supporting application's deployment.
+
+# Motivation
+With the current supporting deployment, our users have to define their application as one of the supporting application's kinds (K8s, Terraform, CloudRun, Lambda, ECS), they can still define multiple PipeCD applications as parts of their application and trigger them one by one, but it's a bit difficult for them to control the deployment flow smoothly.
+
+![image](assets/deployment-flow-kind.png)
+
+Another typical usecase is when users want to build up some kind of deployment flow such as deploy application to the development environment, then if it's done successfully, deploy that application to the staging and then the production environment.
+
+![image](assets/deployment-flow-env.png)
+
+Or a variant of that requirement, when users want to roll out their applications to their cluster one by one based on its region. As the same below image, the order of the rolling out deployment should be flex, supports both sequential and parallel.
+
+![image](assets/deployment-flow-region.png)
+
+All the above requirements share the same thing in the context, that is: it can be done by users' deployment for their applications as PipeCD applications one by one and make trigger those deployments manually via console or make trigger via pull requests to those application configurations separately, but this all manual stub are tedious and difficult to manage smoothly. With this new PipeCD deployment flow feature, all of those manual steps will be replaced, to keep the good point of using a CD system.
+
+# Detailed design
+The idea is to keep the PipeCD application deployment as a unit of deployment as is, but add __a way to enable users to manipulate a deployment flow based on the state of the last sucessfully run deployment__.
+
+A canonical flow would look as below:
+- Users trigger to run their first application (the first application in their deployment flow) via the web console or pull requests as usual (1)
+- Deploy the first triggered application in the deployment flow (2)
+- After the application is deployed successfully, the piped will send a trigger command to trigger the deployment flow to the control-plane, since all those applications are shared in the same project, this trigger should be valid (3)
+- Control-plane gets deployment events triggered by the first application's piped and makes sync deployment commands for all listed applications (4)
+- Pipeds which handle deployment for all triggered applications fetch sync command and deploy its application once the conditions are satisfied (5)
+
+![image](assets/deployment-flow-idea.png)
```

Comment:
Could you change the app name to be something like "app-1" "app-2"?
(Instead of using `terraform` or `lambda` words)